lib/std/jsfetch.nim

```diff
@@ -0,0 +1,152 @@
+
+import asyncjs, jsheaders
+from httpcore import HttpMethod
+
+type
+  FetchOptions* = ref object  ## Options for Fetch API.
+    keepalive*: bool
+    metod* {.importjs: "method".}: cstring
+    body*, integrity*, referrer*, mode*, credentials*, cache*, redirect*, referrerPolicy*: cstring
+
+  FetchModes* = enum  ## JavaScript Fetch API mode options.
+    fmCors = "cors"
+    fmNoCors = "no-cors"
+    fmSameOrigin = "same-origin"
+
+  FetchCredentials* = enum  ## JavaScript Fetch API Credential options.
+    fcInclude = "include"
+    fcSameOrigin = "same-origin"
+    fcOmit = "omit"
+
+  FetchCaches* = enum  ## https://developer.mozilla.org/docs/Web/API/Request/cache
+    fchDefault = "default"
+    fchNoStore = "no-store"
+    fchReload = "reload"
+    fchNoCache = "no-cache"
+    fchForceCache = "force-cache"
+
+  FetchRedirects* = enum  ## JavaScript Fetch API Redirects options.
+    frFollow = "follow"
+    frError = "error"
+    frManual = "manual"
+
+  FetchReferrerPolicies* = enum  ## JavaScript Fetch API Referrer Policy options.
+    frpNoReferrer = "no-referrer"
+    frpNoReferrerWhenDowngrade = "no-referrer-when-downgrade"
+    frpOrigin = "origin"
+    frpOriginWhenCrossOrigin = "origin-when-cross-origin"
+    frpUnsafeUrl = "unsafe-url"
+
+  Response* = ref object  ## https://developer.mozilla.org/en-US/docs/Web/API/Response
+    myBodyUsed*, ok*, redirected*: bool
+    typ* {.importjs: "type".}: cstring
+    url*, statusText*: cstring
+    status*: cint
+    headers*: Headers
+
+
+func unsafeNewFetchOptions*(metod, body, mode, credentials, cache, referrerPolicy: cstring,
+    keepalive: bool, redirect = "follow".cstring, referrer = "client".cstring, integrity = "".cstring): FetchOptions {.importjs:
+    "{method: #, body: #, mode: #, credentials: #, cache: #, referrerPolicy: #, keepalive: #, redirect: #, referrer: #, integrity: #}".}
+  ## **Unsafe** `newfetchOptions`. Low-level func for optimization.
+
+func newfetchOptions*(metod: HttpMethod, body: cstring,
+    mode: FetchModes, credentials: FetchCredentials, cache: FetchCaches, referrerPolicy: FetchReferrerPolicies,
+    keepalive: bool, redirect = frFollow, referrer = "client".cstring, integrity = "".cstring): FetchOptions =
+  ## Constructor for `FetchOptions`.
+  result = FetchOptions(
+    body: body, mode: $mode, credentials: $credentials, cache: $cache, referrerPolicy: $referrerPolicy,
+    keepalive: keepalive, redirect: $redirect, referrer: referrer, integrity: integrity,
+    metod: (case metod
+      of HttpHead:   "HEAD".cstring
+      of HttpGet:    "GET".cstring
+      of HttpPost:   "POST".cstring
+      of HttpPut:    "PUT".cstring
+      of HttpDelete: "DELETE".cstring
+      of HttpPatch:  "PATCH".cstring
+      else:          "GET".cstring
+    )
+  )
+
+func fetchToCstring*(url: cstring): Future[cstring] {.importjs: "fetch(#).then(response => response.text()).then(text => text)".}
```

Comment:
@juancarlospaco can you split this PR into 2:
* lib/std/jsheaders.nim is basically LGTM from me, so can be merged
* lib/std/jsfetch.nim still needs some work, I can explain

Reply:
Explain

Reply:
because `then` should be composable and generic instead of hard-coded as you did here; see https://github.com/nim-lang/Nim/pull/16871 WIP

I modified your PR a bit to use this and make your runnableExample a bit more useful, stay tuned